sync/sync.go

```diff
@@ -71,16 +91,18 @@ type Synchronizer struct {
 	log      utils.SimpleLogger
 	listener EventListener
 
+	pending             atomic.Pointer[Pending]
 	pendingPollInterval time.Duration
 	catchUpMode         bool
 	plugin              junoplugin.JunoPlugin
 }
 
-func New(bc *blockchain.Blockchain, starkNetData starknetdata.StarknetData,
-	log utils.SimpleLogger, pendingPollInterval time.Duration, readOnlyBlockchain bool,
+func New(bc *blockchain.Blockchain, starkNetData starknetdata.StarknetData, log utils.SimpleLogger,
+	pendingPollInterval time.Duration, readOnlyBlockchain bool, database db.DB,
```

Comment:
I think it breaks existing abstractions. But I see that you need to use it in order to generate state for pending block. Maybe it's possible to rely on memory txn ?

Reply:
The DB is not for storing the pending state; it is for accessing the current state and is a read-only operation. 

The pending state is created by fetching the current state and applying the pending StateDiff. To get access to StateReader, we need a DB transaction. So, unfortunately, I cannot replace it with memory txn unless I store the Starknet state in memory (which is not a good idea).

Reply:
got it, thanks for clarifications